File: Data Structures & Algorithms/counting-bits/submission-8.cs
public class Solution
{
    public int[] CountBits(int n)
    {
        var res = new int[n + 1];
        while (n > 0)
        {
            int num = n;
            while (num != 0) {
                res[n]++;
                num &= (num - 1);
            }
            n--;
        }

        return res;
    }
}

File: Data Structures & Algorithms/top-k-elements-in-list/submission-8.py

class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        d = {}

        for num in nums:
            
            if num not in d:
                d[num] = 1
            else:
                d[num] += 1

        items = list(d.items())
        n = len(items)
        for i in range(n):
            for j in range(n - 1 - i):
                if items[j][1] < items[j + 1][1]:
                    items[j], items[j + 1] = items[j + 1], items[j]

        return list(dict(items).keys())[:k]
